test/conftest.py

```diff
@@ -121,3 +124,13 @@ def fake_time(monkeypatch):
     time = FakeTime()
     monkeypatch.setattr(util, "monotonic_time", time.monotonic_time)
     return time
+
+
+@pytest.fixture(scope="module")
+def srv_factory():
+    def _factory(path):
+        cfg = config.load(path)
+        cfg.local.socket = f"\0/org/ovirt/imageio/{uuid.uuid4()}"
+        return server.Server(cfg)
+
+    return _factory
```

Comment:
Why do we need a factory?

See how the nbd_server fixture works. It returns an object
that can be configured for a particular test, and then started.
When fixture is cleaned up it stops the server. This is similar
to other pytest fixtures.

I want to avoid the mess when every fixture behaves in a different
way. Sometimes its a string, sometimes a function, sometimes an
object (e.g. tmpdir). Lets keep all fixtures as objects. If configuration
is needed, set the object attributes or call methods of the object.

But I'm not sure we need a fixture for the server - the thing that
need to change in many tests is only the config, so maybe we need
a helper function to load a test config:

    def load_config(path):
        cfg = config.load(path)
        cfg.local.socket = f"\0/org/ovirt/imageio/{uuid.uuid4()}"
        return cfg

Can be in testutil.py.